freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/provision/action/FreeIpaProvisionActions.java

```diff
@@ -90,12 +92,29 @@ protected Selectable createRequest(StackContext context) {
         };
     }
 
+    @Bean(name = "CLUSTERPROXY_UPDATE_REGISTRATION_STATE")
+    public Action<?, ?> updateClusterProxyRegistrationAction() {
+        return new AbstractStackProvisionAction<>(InstallFreeIpaServicesSuccess.class) {
+            @Override
+            protected void doExecute(StackContext context, InstallFreeIpaServicesSuccess payload, Map<Object, Object> variables) throws Exception {
+                stackUpdater.updateStackStatus(context.getStack().getId(), DetailedStackStatus.UPDATE_CLUSTER_PROXY_REGISTRATION,
+                        "Updating cluster proxy registration.");
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(StackContext context) {
+                return new ClusterProxyUpdateRegistrationRequest(context.getStack().getId());
+            }
+        };
+    }
+
     @Bean(name = "FREEIPA_POST_INSTALL_STATE")
     public Action<?, ?> postInstallFreeIpa() {
-        return new AbstractStackProvisionAction<>(InstallFreeIpaServicesSuccess.class) {
+        return new AbstractStackProvisionAction<>(ClusterProxyUpdateRegistrationSuccess.class) {
```

Comment:
we could use `StackEvent` here so both the old and new event would match

Reply:
The `StackEvent` is still used during the original bootstrapping for the `StackProvisionFlowConfig`. The `ClusterProxyUpdateRegistrationSuccess` is for the new step in the `FreeIpaProvisionFlowConfig`. The `StackProvisionFlowConfig` events remain unchanged. Does. it still make sense to have a `StackEvent` here? If so I will make the change.